src/contexts/ErrorContext/ErrorContext.tsx

```diff
@@ -1,85 +1,69 @@
 import Prompt from "components/Prompt";
-import { GENERIC_ERROR_MESSAGE } from "constants/common";
-import { APError, AP_ERROR_DISCRIMINATOR } from "errors/errors";
+import { EMAIL_SUPPORT } from "constants/env";
 import { logger } from "helpers";
-import {
-  PropsWithChildren,
-  createContext,
-  useCallback,
-  useContext,
-} from "react";
+import { ReactNode } from "react";
 import { useModalContext } from "../ModalContext";
 
-type State = {
-  handleError: (
-    error: any,
-    displayMessage?: string,
-    options?: { log: boolean }
-  ) => void;
-};
+function parseError(error: unknown): string | undefined {
+  if (typeof error === "string") return error;
 
-const Context = createContext<State>({
-  handleError: (_: any, _1?: string, _2?: { log: boolean }) => {},
-});
+  if (typeof error === "object" && error != null) {
+    if ("message" in error) return parseError(error.message);
+    if ("data" in error) return parseError(error.data);
+    if ("error" in error) return parseError(error.error);
+  }
 
-export default function ErrorContext(props: PropsWithChildren<{}>) {
+  if (error instanceof Error) return error.message;
+}
+
+export function useErrorContext() {
   const { showModal } = useModalContext();
 
   /**
-   * biome-ignore lint/correctness/useExhaustiveDependencies: bug in Biome with recursive dependencies
-   * Created an issue on their GH, see https://github.com/biomejs/biome/issues/2361
+   * @description for expected errors
+   * @param message - user can do something about
+   * @param error - if provided, would be logged to sentry for further investigation
    */
-  const handleError: State["handleError"] = useCallback(
-    (error: any, displayMessage?: string, options = { log: true }) => {
-      if (options.log) {
-        logger.error(error);
-      }
+  function displayError(message: ReactNode, error?: unknown) {
+    showModal(Prompt, { type: "error", children: message });
+    if (error) {
+      logger.error(error);
+    }
+  }
 
-      if (displayMessage) {
-        showModal(Prompt, {
-          type: "error",
-          children: displayMessage,
-        });
-      } else if (typeof error === "string") {
-        showModal(Prompt, {
-          type: "error",
-          children: error,
-        });
-      } else if (instanceOfAPError(error.data)) {
-        handleError(error.data);
-      } else if ("message" in error) {
-        handleError(error.message);
-        //TODO: specify controlled error shapes
-      } else if (error.data && typeof error.data === "string") {
-        handleError(error.data);
-      } else if (error.data && "message" in error.data) {
-        handleError(error.data.message);
-      } else if ("error" in error) {
-        handleError(error.error);
-      } else {
-        showModal(Prompt, {
-          type: "error",
-          children: GENERIC_ERROR_MESSAGE,
-        });
-      }
-    },
-    [showModal]
-  );
+  type Generic = {
```

Comment:
nit: `Generic` and `DisplayType` can be defined outside the function.

Reply:
yes could be, but I like the idea that these types can't be used outside of its lexical scope. Here's another [example](https://github.com/AngelProtocolFinance/angelprotocol-web-app/blob/8748f2943b5e3a3ddf793e132899d90b1fd41d31/src/pages/Admin/Charity/EditProfile/useEditProfile.ts#L67), the type is closer to where it is used.